nixos/modules/programs/wayland/hyprland.nix

```diff
@@ -63,7 +63,7 @@ in
 
       systemd = lib.mkIf cfg.systemd.setPath.enable {
         user.extraConfig = ''
-          DefaultEnvironment="PATH=$PATH:/run/current-system/sw/bin:/etc/profiles/per-user/%u/bin:/run/wrappers/bin"
+          DefaultEnvironment="PATH=/run/wrappers/bin:/etc/profiles/per-user/%u/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin:$PATH"
```

Comment:
Is there a reason to prioritise `etc/profiles/per-user` over current system's bin?

Reply:
No. Should I put `/etc/profiles/per-user` at the end, before `$PATH`?

Reply:
This order is fine, user-specific binaries should take priority.
Run `printenv PATH` in a shell.

But the existence of this option in the first place is strange and defeats the intentional isolation that systemd provides.

Reply:
This option was rolled out specifically to mitigate a Hyprland bug where opening apps using `xdg-portal` service was not possible. Clicking discord/telegram links from a browser would fail to open the telegram app, without this option. This was applicable for every apps.

This was not done without the *current usecase* in mind.

Another workaround for this was to run
`dbus-update-activation-environment --systemd --all` activation which would would import all system variables, including PATH, or just selectively import PATH using systemctl.

Reply:
`dbus-update-activation-environment --systemd` is the normal solution to this problem, it's what I use on sway, and importing PATH was not necessary

Reply:
It would likely go in the compositor-specific config file, so factoring it out into a common file may not make sense. e.g. for sway:
https://github.com/NixOS/nixpkgs/blob/dce10c76c491a4bffce0247ea6c33345ab1e6c72/nixos/modules/programs/wayland/sway.nix#L125

Also `SWAYSOCK` is sway-specific ofc

Reply:
That looks like a sway's own config way to handle this. That kind of approach is certainly doable using Hyprland's own configs, but that's not what I am asking for. 

Ideally this should be really handled on the system level, perhaps by running some commands at user login.

Reply:
> but that's not what I am asking for.

as I said, I doubt it can be factored out into a common file

Reply:
Just to let you guys know, Hyprland already has integration with systemd and dbus by default, see: https://github.com/hyprwm/Hyprland/blob/c1e21719a2fff2fa9549f00053ac40173da54af9/src/config/ConfigManager.cpp#L1288

And for me, using flatpak with `xdg.portal.xdgOpenUsePortal = true;` and `programs.hyprland.systemd.setPath.enable = false;`, Discord can open links on Firefox and invite links also work from Firefox to the Discord client. But maybe other configs can be affecting this issue, I'm not sure.

Reply:
In any case I believe the concern with this PR was addressed, and our discussion is more abilt future improvements.